src/Symfony/Bridge/Twig/Extension/IntlExtension.php

```diff
@@ -0,0 +1,297 @@
+     * @param string|null               $displayLocale
+     * @param \DateTimeZone|string|null $timezone
+     * @param string|null               $format
+     * @param string                    $calendar
+     *
+     * @return bool|string
+     *
+     * @throws \Exception|\InvalidArgumentException If the date or a format is invalid
+     */
+    public function getLocalizedDate($date, $dateFormat = 'medium', $timeFormat = 'medium', $displayLocale = null, $timezone = null, $format = null, $calendar = 'gregorian')
+    {
+        if (is_string($date)) {
+            $date = new \DateTime($date);
+        }
+
+        if (!$date instanceof \DateTimeInterface) {
+            throw new \InvalidArgumentException(sprintf('Expected a \DateTimeInterface or time string got "%s"', is_object($date) ? get_class($date) : gettype($date)));
+        }
+
+        if (false !== $timezone) {
+            if (!$timezone instanceof \DateTimeZone) {
+                $timezone = new \DateTimeZone($timezone ?: date_default_timezone_get());
+            }
+            $date->setTimezone($timezone);
+        }
+
+        $formatter = $this->createDateFormatter(
+            $displayLocale ?: $this->defaultLocale,
+            $dateFormat,
+            $timeFormat,
+            $timezone->getName(),
+            'gregorian' === $calendar ? IntlDateFormatter::GREGORIAN : IntlDateFormatter::TRADITIONAL,
+            $format
+        );
+
+        return $formatter->format($date->getTimestamp());
+    }
+
+    /**
+     * Returns a localized language name for a given code.
+     *
+     * @param string      $language
+     * @param string|null $displayLocale
+     */
+    public function getLocalizedLanguage($language, $displayLocale = null)
+    {
+        $separator = false === strpos($language, '-') ? '_' : '-';
+        $params = explode($separator, (string) $language, 2);
+        $lang = $params[0];
+        $region = isset($params[1]) ? $params[1] : null;
+
+        return Intl::getLanguageBundle()->getLanguageName($lang, $region, $displayLocale ?: $this->defaultLocale);
+    }
+
+    /**
+     * Returns a localized locale name for a given code.
+     *
+     * @param string      $locale
+     * @param string|null $displayLocale
+     */
+    public function getLocalizedLocale($locale, $displayLocale = null)
+    {
+        return Intl::getLocaleBundle()->getLocaleName((string) $locale, $displayLocale ?: $this->defaultLocale);
+    }
+
+    /**
+     * Returns a number formatted for the given locale and options.
+     * 
+     * @param mixed  $number
+     * @param string $style
+     * @param string $type
+     * @param null   $displayLocale
+     *
+     * @return bool|string
+     *
+     * @throws \InvalidArgumentException When the type or the style is invalid
+     */
+    public function getLocalizedNumber($number, $style = 'decimal', $type = 'default', $displayLocale = null)
+    {
+        if (empty($this->numberTypeValues)) {
```

Comment:
if we really need this lazy initialization, can't we use `null !== $this->numberTypeValues` (and have no default value on the property?)